src/NUnitFramework/nunitlite.tests/TextUITests.cs

```diff
@@ -539,6 +539,28 @@ public void TwoTests_NestedExecution(string labels, string expected)
             Assert.That(Report, Is.EqualTo(expected));
         }
 
+        [TestCase("text", "stream", "testId", "testName")]
+        [TestCase("text", "stream", null, "testName")]
+        [TestCase("text", "stream", "testId", null)]
+        public void TestOutputToXml_IncludeAttributesInProperFormatting(string text, string stream, string testId, string testName)
+        {
+            var testOutput = new TestOutput(text, stream, testId, testName);
+            var expected = new StringBuilder();
+            expected.AppendFormat("<test-output stream=\"{0}\"", stream);
+
+            if (testId != null)
+            {
+                expected.AppendFormat(" testid=\"{0}\"", testId);
+            }
+
+            if (testName != null)
+            {
+                expected.AppendFormat(" testname=\"{0}\"", testName);
+            }
+
+            expected.AppendFormat("><![CDATA[{0}]]></test-output>", text);
+            Assert.That(testOutput.ToXml(), Is.EqualTo(expected.ToString()));
+        }
```

Comment:
It's a good test, but it's a low-level test of `TestOutput` not of `TextUI`. In addition, `TextUI` is a bad place because it's part of the `nunitlite` assembly, not of `nunit.framework`.

Best choice is to create a new class for `TestOutputTests` with this single test in it. Second, acceptable to me, choice is to piggyback it on `TextOutputTests` because that's at least in the correct assembly. 😄